### ISSUE TITLE
oxlint: override application drops Vitest-compatible jest/* category rules when any override matches

### ISSUE THREAD

**Issue by therockstorm** [oxc-project/oxc#21284]
## What version of Oxlint are you using?

```text
1.59.0
```

## What command did you run?

```text
oxlint --deny-warnings
```

## What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "categories": {
    "style": "error"
  },
  "plugins": ["vitest"],
  "rules": {},
  "overrides": [
    {
      "files": ["**/*.spec.ts"],
      "rules": {}
    }
  ]
}
```

## What happened?

I found what looks like a bug in override application for Vitest configs.

### Summary

When the `vitest` plugin is enabled, Oxlint intentionally includes some `jest/*` rules for Vitest compatibility. However, if any override matches a file, the override application path appears to forget that `vitest` implies those compatible `jest/*` rules.

As a result, an override with an empty `rules` object can change lint results for matched test files, even though an empty override should be a no-op.

### Minimal reproduction

1. Create a project with at least one `*.spec.ts` file containing:
   - a `describe(...)` block
   - an `it(...)` block
   - a `toEqual(...)` assertion
2. Use this config:

```jsonc
{
  "categories": { "style": "error" },
  "plugins": ["vitest"],
  "rules": {},
  "overrides": []
}
````

3. Run:

```bash
oxlint --deny-warnings
```

Observed:

* `jest/*` errors are reported on the `*.spec.ts` file, for example:

  * `jest/prefer-strict-equal`
  * `jest/prefer-lowercase-title`

4. Now add an override matching the same test files, but leave `rules` empty:

```jsonc
{
  "categories": { "style": "error" },
  "plugins": ["vitest"],
  "rules": {},
  "overrides": [
    {
      "files": ["**/*.spec.ts"],
      "rules": {}
    }
  ]
}
```

5. Run the same command again:

```bash
oxlint --deny-warnings
```

Observed:

* The `jest/*` errors on matched `*.spec.ts` files disappear
* `vitest/*` errors do not disappear
* Other non-jest rule counts stay effectively unchanged

Expected behavior:

An override with `"rules": {}` should be a no-op.
It should not suppress any rules for matched files.
The `jest/*` error count should be identical with and without the empty override.

### Observed behavior in a larger repo

Using `"style": "error"` in a monorepo with about 380 files:

* No override:

  * `jest/*` errors: 512
  * `vitest/*` errors: 429
  * all other errors: about 3,875

* Empty `**/*.spec.ts` override:

  * `jest/*` errors: 68
  * `vitest/*` errors: 429
  * all other errors: about 3,887

Interpretation:

* `jest/*` errors dropped by 444
* the remaining 68 were from `*.test.ts` files not matched by the `*.spec.ts` glob
* `vitest/*` errors were unchanged
* the effect appears specific to `jest/*` compatibility rules under Vitest

### Why I think this is happening

I inspected the Oxlint source and found a likely inconsistency between the builder path and the override-application path.

In `crates/oxc_linter/src/config/config_builder.rs`:

* `get_all_rules_for_plugins()` explicitly expands `vitest` to include `jest`
* `build()` also explicitly expands `vitest` to include `jest`

That means the base config intentionally treats Vitest as enabling a compatibility subset of Jest rules.

However, in `crates/oxc_linter/src/config/config_store.rs`, `Config::apply_overrides()` rebuilds the active rule map for matched files and filters `self.base_rules` and `RULES` using `plugins` directly, without applying the same `vitest -> jest` compatibility expansion.

So when any override matches, Oxlint takes the `apply_overrides()` path, rebuilds rules from the raw plugin set, and drops the compatible `jest/*` rules that were present in the base rule set for Vitest.

That would explain why:

* base config shows `jest/*` compatibility rules for Vitest
* any matching override causes those `jest/*` rules to disappear
* an empty override is enough to reproduce the bug

### Likely code locations

1. `crates/oxc_linter/src/config/config_builder.rs`

   * `get_all_rules_for_plugins()`
   * `build()`
     These already do:
   * if plugins contain `VITEST`, also include `JEST`

2. `crates/oxc_linter/src/config/config_store.rs`

   * `Config::apply_overrides()`
     This currently rebuilds rules using `plugins` directly and appears to miss the same compatibility expansion

### Proposed fix

Inside `Config::apply_overrides()`, compute an `effective_plugins` set before filtering rules, similar to the builder path:

```rust
let mut effective_plugins = plugins;
if effective_plugins.contains(LintPlugins::VITEST) {
    effective_plugins |= LintPlugins::JEST;
}
```

Then use `effective_plugins` instead of `plugins` for:

1. filtering `self.base_rules`
2. building `all_rules`
3. initializing `configured_plugins`, so root Vitest configs also count Jest compatibility as already configured

In other words, the override application path should use the same effective plugin semantics as the builder path.

### Suggested regression tests

Please add tests in `crates/oxc_linter/src/config/config_store.rs` covering at least these cases:

1. Root config:

   * plugins: `vitest`
   * category enabling a rule category that includes compatible `jest/*` rules
   * no overrides
   * assert that a compatible `jest/*` rule is active

2. Same config plus a matching override with:

   * `files: ["**/*.spec.ts"]`
   * `rules: {}`
   * assert that the same compatible `jest/*` rule is still active

3. Same config plus a matching override with explicit empty plugin override omitted:

   * `plugins` not set
   * confirm behavior is unchanged from base

4. Optional:

   * override introduces `vitest` rather than root config doing so
   * confirm category-enabled compatible `jest/*` rules are added there too

A good assertion target would be a known Vitest-compatible Jest rule such as:

* `jest/prefer-strict-equal`
* `jest/prefer-lowercase-title`

### Additional context

This may be related to existing Jest/Vitest plugin confusion, but I think this is a distinct bug in override application:

* `config_builder.rs` clearly intends `vitest` to imply compatible `jest/*` rules
* `apply_overrides()` does not preserve that behavior when an override matches
* existing "empty override is a no-op" tests do not seem to cover Vitest/Jest compatibility rules, so this regression path is missed

**Comment by camc314** [comment]
fixed by #21389
